craft_application/models/manifest.py

```diff
@@ -63,6 +79,27 @@ def marshal(self) -> dict[str, str | list[str] | dict[str, Any]]:
             exclude_defaults=False,  # to include 'metadata_generator'
         )
 
+    @staticmethod
+    def from_packed_artifact(
+        project: models.Project, build_info: models.BuildInfo, artifact: pathlib.Path
+    ) -> "ProjectManifest":
+        """Create the project manifest for a packed artifact."""
+        hashes = Hashes.from_path(artifact)
+
+        now = datetime.now(timezone.utc)
+
+        return ProjectManifest.unmarshal(
+            {
+                "component-name": project.name,
+                "component-version": project.version,
+                "component-description": project.summary,
+                "component-id": {"hashes": hashes.marshal()},
+                "architecture": build_info.build_for,
+                "license": project.license,
+                "creation_timestamp": now.isoformat(),
+            }
+        )
```

Comment:
Similar to above, since this is a constructor

```suggestion
    @classmethod
    def from_packed_artifact(
        cls, project: models.Project, build_info: models.BuildInfo, artifact: pathlib.Path
    ) -> Self:
        """Create the project manifest for a packed artifact."""
        hashes = Hashes.from_path(artifact)

        now = datetime.now(timezone.utc)

        return cls.unmarshal(
            {
                "component-name": project.name,
                "component-version": project.version,
                "component-description": project.summary,
                "component-id": {"hashes": hashes.marshal()},
                "architecture": build_info.build_for,
                "license": project.license,
                "creation_timestamp": now.isoformat(),
            }
        )
```